backfill.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# this script is used to update the article-json stored in lax
+# performing just an INGEST of content, *no* PUBLISH events are
+# sent whatsoever.
+
+# talking to lax via the adaptor.py script for many thousands of articles is
+# extremely slow, so this script bypasses all that, bulk generates articles,
+# bulk validates them then tells lax to do a bulk ingest.
+
+set -euo pipefail # strict mode
+#set -xv # debugging
+
+# housekeeping
+
+trap ctrl_c INT
+function ctrl_c() {
+    echo "caught ctrl-c"
+    exit 1
+}
+
+srcdir=$(pwd)
+errcho(){ >&2 echo $@; }
+
+# where articles will be linked to/downloaded for backfill
+defaultrunpath="run-$(date +'%Y%m%d%H%M%S')"
+runpath=${1:-$defaultrunpath}
+runpath=$(realpath $runpath) # realpath needs to be installed :(
+mkdir -p "$runpath"
+
+shift # download-api-raml.sh takes an arg :P
```

Comment:
then we should use `sh` instead of `source` to create a subprocess rather than include the other script? overhead is minimal and isolation should prevent these problems